Ironmon-Tracker.lua

```diff
@@ -14,56 +19,42 @@ if client.getversion == nil or (client.getversion() ~= "2.8" and client.getversi
 	return
 end
 
--- Root folder for the project data and sub scripts
-DATA_FOLDER = "ironmon_tracker"
-
--- Get the user settings saved on disk and create the base Settings object
-INI = dofile(DATA_FOLDER .. "/Inifile.lua")
--- Need to manually read the file to work around a bug in the ini parser, which
--- does not correctly handle that the last iteration over lines() returns nil
-local file = io.open("Settings.ini")
-if file ~= nil then
-	Settings = INI.parse(file:read("*a"), "memory")
-	io.close(file)
-end
-
 -- Import all scripts before starting the main loop
-dofile(DATA_FOLDER .. "/Constants.lua")
-dofile(DATA_FOLDER .. "/PokemonData.lua")
-dofile(DATA_FOLDER .. "/MoveData.lua")
-dofile(DATA_FOLDER .. "/MiscData.lua")
-dofile(DATA_FOLDER .. "/Memory.lua")
-dofile(DATA_FOLDER .. "/GameSettings.lua")
-dofile(DATA_FOLDER .. "/InfoScreen.lua")
-dofile(DATA_FOLDER .. "/Options.lua")
-dofile(DATA_FOLDER .. "/Theme.lua")
-dofile(DATA_FOLDER .. "/ColorPicker.lua")
-dofile(DATA_FOLDER .. "/Utils.lua")
-dofile(DATA_FOLDER .. "/TrackerScreen.lua")
-dofile(DATA_FOLDER .. "/Input.lua")
-dofile(DATA_FOLDER .. "/Drawing.lua")
-dofile(DATA_FOLDER .. "/Program.lua")
-dofile(DATA_FOLDER .. "/Pickle.lua")
-dofile(DATA_FOLDER .. "/Tracker.lua")
-
-Main = {}
-Main.LoadNextSeed = false
+dofile(Main.DataFolder .. "/Inifile.lua")
+dofile(Main.DataFolder .. "/Constants.lua")
+dofile(Main.DataFolder .. "/PokemonData.lua")
+dofile(Main.DataFolder .. "/MoveData.lua")
+dofile(Main.DataFolder .. "/MiscData.lua")
+dofile(Main.DataFolder .. "/Memory.lua")
+dofile(Main.DataFolder .. "/GameSettings.lua")
+dofile(Main.DataFolder .. "/InfoScreen.lua")
+dofile(Main.DataFolder .. "/Options.lua")
+dofile(Main.DataFolder .. "/Theme.lua")
+dofile(Main.DataFolder .. "/ColorPicker.lua")
+dofile(Main.DataFolder .. "/Utils.lua")
+dofile(Main.DataFolder .. "/TrackerScreen.lua")
+dofile(Main.DataFolder .. "/Input.lua")
+dofile(Main.DataFolder .. "/Drawing.lua")
+dofile(Main.DataFolder .. "/Program.lua")
+dofile(Main.DataFolder .. "/Pickle.lua")
+dofile(Main.DataFolder .. "/Tracker.lua")
 
 -- Displays a given error message in a pop-up dialogue box
 function Main.DisplayError(errMessage)
 	client.pause()
-	local form = forms.newform(400, 130, "[v" .. TRACKER_VERSION .. "] Woops, there's been an error!", function() return end)
+	local form = forms.newform(400, 130, "[v" .. Main.TrackerVersion .. "] Woops, there's been an error!", function() return end)
 	Utils.setFormLocation(form, 100, 50)
-	forms.label(form, errMessage,  18, 10, 400, 50)
+	forms.label(form, errMessage, 18, 10, 400, 50)
 	forms.button(form, "Close", function()
 		client.unpause()
 		forms.destroy(form)
 	end, 155, 70)
-	return
 end
 
 -- Main loop
 function Main.Run()
+	Main.LoadSettings()
```

Comment:
This function returns true/false for if the settings are loaded correctly, however this function call here doesn't seem to make use of it? 

Seems like the tracker will just attempt to continue running as normal even if the function here doesn't successfully load the settings (returns false)

Reply:
Ah, this is true. I included the returns as an optionally way to use `LoadSettings`, in case we wanted that in the future. THey aren't needed now, as there is enough error checking and safety nets to prevent anything bad from happening during the loading of the settings. As such, I can simply remove the `return`s, but also don't think they're hurting anything by staying in?